sdcm/cluster.py

```diff
@@ -2548,60 +2548,27 @@ def get_console_screenshot(self):
-        # means resharding was started
-
-        return bool(found) if status == 'start' else not bool(found)
-
-    # Default value of murmur3_partitioner_ignore_msb_bits parameter is 12
-    def _restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits: int = 12):
         self.stop_scylla()
         # Change murmur3_partitioner_ignore_msb_bits parameter to cause resharding.
         with self.remote_scylla_yaml() as scylla_yml:
             scylla_yml.murmur3_partitioner_ignore_msb_bits = murmur3_partitioner_ignore_msb_bits
-        search_reshard = self.follow_system_log(patterns=['Reshard', 'Reshap'])
+
+        search_reshard = self.follow_system_log(patterns=[DB_LOG_PATTERN_RESHARDING_START])
         self.start_scylla(timeout=7200)
-        return search_reshard
+        resharding_started = list(search_reshard)
 
-    def restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits: int = 12) -> None:
-        # TODO: reuse 'DB_LOG_PATTERN_RESHARDING_START' and 'DB_LOG_PATTERN_RESHARDING_FINISH' vars here.
-        search_reshard = self._restart_node_with_resharding(murmur3_partitioner_ignore_msb_bits)
-
-        resharding_started = wait.wait_for(func=self._resharding_status, step=5, timeout=180,
-                                           text="Wait for re-sharding to be started", status='start', throw_exc=False)
-        if not resharding_started:
-            resharding_started = list(search_reshard)
-            if resharding_started:
-                # If re-sharding was found running in the log but didn't reported by "nodetool compactionstats",
-                # it means that re-sharding was started and finished already
-                self.log.debug(f'Resharding has been finished successfully '
-                               f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits})')
-            else:
-                raise Exception(f'Resharding has not been started '
-                                f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
-                                'Check the log for the details')
-            return
-        else:
-            # Decrease nodetool compactionstats calls from 5sec to 1min to avoid the noise
-            resharding_finished = wait.wait_for(func=self._resharding_status, step=60, throw_exc=False,
-                                                text="Wait for re-sharding to be finished", status='finish')
-
-            if not resharding_finished:
-                raise Exception('Resharding was not finished! '
-                                f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
-                                'Check the log for the details')
-            self.log.debug('Resharding has been finished successfully '
+        if resharding_started:
+            self.log.debug(f'Resharding has been finished successfully '
```

Comment:
This is not accurate, here it says resharding finished successfully, but we don't know it here, we just know it started.

Reply:
But Scylla startup will failed to finish (timeout that we set is 3600) and raise exception if resharding won't finish. So I suppose here that that resharding finished as Scylla startup completed

Reply:
you are right the timeout would catch if the resharding won't finish in time.
but what about the resharding process failing ? (I'm guessing there could be things that would casue it to fail)

can we look for both strings ? the start and the end with success ?